design/validator_info.md

```diff
@@ -0,0 +1,163 @@
+# Extension of validator_info
+The document contains list of possible extensions and modifications of validator_info command to simplify node state monitoring and debug process
+
+* [Validator_info Description](#cur-description)
+* [Modification - New Read Command](#new-command)
+* [Extension - Additional Fields](#new-fields)
+
+## Validator_info Description
+Validator_info is a script that formats and prints data provided by node. The script should be run manually by
+the user who has a read access to the \<node name\>_info.json file from the node's data directory.
+This file is updated by node once a minute and contains following information:
+```
+{
+    "did": "Gw6pDLhcBcoQesN72qfotTgFa7cbuqZpkX3Xo6pLhPhv", # node's identidier
+    "response-version": "0.0.1", # 0.0.1 for now
+    "timestamp": 1519711338, # current time 
+    "verkey": "33nHHYKnqmtGAVfZZGoP8hpeExeH45Fo8cKmd5mcnKYk7XgWNBxkkKJ", # node's verkey
+    "pool": { # current pool
+        "reachable": { # reachable nodes
+            "list": ["Node1", "Node2", "Node3", "Node4"],
+            "count": 4
+         },
+        
+        "total-count": 4, # total count of nodes
+        "unreachable": { # unreachable nodes
+           "list": [],
+           "count": 0
+        }
+     },
+    
+    "bindings": { # node's network configuration
+       "client": {
+            "port": 9702,
+            "protocol": "tcp"
+        },
+       
+       "node": {
+            "port": 9701,
+            "protocol": "tcp"
+        }
+    },
+    
+    "metrics": { # some numeric characteristics
+        "transaction-count": { # txn count by ledger
+            "pool": 4,
+            "ledger": 19,
+            "config": 0
+        },
+        
+        "average-per-second": { # performance counters
+            "write-transactions": 0.013294133790137249,
+            "read-transactions": 0.0
+        },
+        
+        "uptime": 300 # uptaime
+    },
+    
+    "software": { # packets' versions
+        "indy-node": "1.3.319",
+        "sovrin": null
+    },
+    
+    "alias": "Node1" # node's name
+}
+```
+
+## Modification - New Read Command
+Validator_info should be accessible as read command, available for TRUSTEE only. New command should provide info from
```

Comment:
I think it should be accessible by STEWARDS as well

Reply:
changed